addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java

```diff
@@ -0,0 +1,386 @@
+                        new QuantityType<>(bdTempExhaust, SIUnits.CELSIUS));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_TEMPERATURE_INCOMING_BEFORE_HEATING,
+                        new QuantityType<>(bdTempIncomingBeforeHeating, SIUnits.CELSIUS));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_TEMPERATURE_INCOMING,
+                        new QuantityType<>(bdTempIncoming, SIUnits.CELSIUS));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_HUMIDITY,
+                        new QuantityType<>(bdHumidity, SmartHomeUnits.PERCENT));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_CELLSTATE, new DecimalType(bdCellstate));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_UPTIME_YEARS, new DecimalType(bdUptimeYears));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_UPTIME_HOURS, new DecimalType(bdUptimeHours));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_UPTIME_HOURS_CURRENT,
+                        new DecimalType(bdUptimeHoursCurrent));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_EXTR_FAN_BALANCE_BASE,
+                        new QuantityType<>(bdExtrFanBalanceBase, SmartHomeUnits.PERCENT));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_SUPP_FAN_BALANCE_BASE,
+                        new QuantityType<>(bdSuppFanBalanceBase, SmartHomeUnits.PERCENT));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_HOME_SPEED_SETTING,
+                        new QuantityType<>(bdHomeSpeedSetting, SmartHomeUnits.PERCENT));
+                updateChannel(ValloxMVBindingConstants.CHANNEL_HOME_AIR_TEMP_TARGET,
+                        new QuantityType<>(bdHomeAirTempTarget, SIUnits.CELSIUS));
+
+                voHandler.updateStatus(ThingStatus.ONLINE);
+                voHandler.dataUpdated();
+            } catch (IOException e) {
+                connectionError(e);
+            } catch (Exception e) {
+                logger.error("Error: {}", e.getMessage());
+            }
+            logger.debug("Print final");
+        }
+
+        private void updateChannel(String strChannelName, State state) {
+            Channel channel = voHandler.getThing().getChannel(strChannelName);
+            if (channel != null) {
+                voHandler.updateState(channel.getUID(), state);
+            }
+        }
+
+        private BigDecimal getNumber(byte[] bytes, int pos) {
+            return (new BigDecimal(bytes[pos] & 0xff)).multiply(new BigDecimal(256))
+                    .add(new BigDecimal(bytes[pos + 1] & 0xff));
+        }
+
+        private BigDecimal getTemperature(byte[] bytes, int pos) {
+            // Fetch 2 byte number out of bytearray representing the temperature in milli degree kelvin
+            BigDecimal bdTempMiliKelvin = getNumber(bytes, pos);
+            // Return number converted to degree celsius
+            return bdTempMiliKelvin.divide(new BigDecimal(100))
```

Comment:
Can the unit framework maybe help us in doing this?

Reply:
here we go:

```
new QuantityType<>(bdTempMiliKelvin, MetricPrefix.MILLI(SmartHomeUnits.KELVIN)).toUnit(SIUnits.CELSIUS);
```